clicache/src/CacheTransactionManager.cpp

```diff
@@ -35,14 +35,7 @@ namespace Apache
       {
         _GF_MG_EXCEPTION_TRY2
 
-          try
-          {
-            m_nativeptr->get()->begin( );
-          }
-          finally
-          {
-            GC::KeepAlive(m_nativeptr);
-          }
+          m_nativeptr->begin( );
```

Comment:
Do you still need the keepalive here? This CLI object can collect while still in the native begin method. Does the CLI object own and destroy the native object?